sdb/commands/zfs/dbuf.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import sdb
+
+
+class DslDir():
+
+    @staticmethod
+    def name(dd):
+        pname = ''
+        if dd.dd_parent:
+            pname = DslDir.name(dd.dd_parent) + '/'
+        return pname + dd.dd_myname.string_().decode("utf-8")
+
+
+class Dataset():
+
+    @staticmethod
+    def name(ds):
+        if not ds:
+            return 'MOS'
+        name = DslDir.name(ds.ds_dir)
+        if not ds.ds_prev:
+            sn = ds.ds_snapname.string_().decode("utf-8")
+            if len(sn) == 0:
+                sn = '%UNKNOWN_SNAP_NAME%'
+            name += '@' + sn
+        return name
+
+
+class Objset():
+
+    @staticmethod
+    def name(os):
+        return Dataset.name(os.os_dsl_dataset)
+
+
+class Dbuf(sdb.Locator, sdb.PrettyPrinter):
+
+    names = ["dbuf"]
```

Comment:
For some other commands in the "zfs" directory, we've prefixed the command with `zfs_`.. should we do that here?

I don't feel strongly either way, but this does make me think that perhaps we should add some sort of command name namespace. E.g. we could specify `dbuf` here like you're doing, and automatically create a `zfs:dbuf` alias. Then, if there's a name conflict with some other directory/module also creating a `dbuf` command, then we could force the use of the more specific name `zfs:dbuf`. While this probably isn't a practical issue for "dbuf" specifically, but I could see it being a problem for a "list" command.

I don't think there's anything to be done in the context of this PR, but just wanted to bring up the idea, in case others have thought about this.

Reply:
I'm not a big fan of the `zfs_` prefix except when it is really needed. I think for `dbufs` we should be safe.

I like the automatic `zfs:` idea and I think we should open an issue for it. We should also open an issue for detecting naming conflicts and link it to it. Then we can have all these discussions separately from this review.

Reply:
I opened: https://github.com/delphix/sdb/issues/77